benchmarks/queue/src/main.rs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use clap::Parser;
+use iceoryx2_bb_lock_free::spsc::index_queue::FixedSizeIndexQueue;
+use iceoryx2_bb_lock_free::spsc::queue::Queue;
+use iceoryx2_bb_lock_free::spsc::safely_overflowing_index_queue::FixedSizeSafelyOverflowingIndexQueue;
+use iceoryx2_bb_posix::barrier::*;
+use iceoryx2_bb_posix::clock::Time;
+use iceoryx2_bb_posix::thread::ThreadBuilder;
+
+const ITERATIONS: u64 = 100000000;
+
+trait PushPop: Send + Sync {
+    fn push(&self, value: usize);
+    fn pop(&self) -> Option<usize>;
+}
+
+impl<const CAPACITY: usize> PushPop for Queue<usize, CAPACITY> {
+    fn push(&self, value: usize) {
+        unsafe { self.push(&value) };
+    }
+
+    fn pop(&self) -> Option<usize> {
+        unsafe { self.pop() }
+    }
+}
+
+impl<const CAPACITY: usize> PushPop for FixedSizeIndexQueue<CAPACITY> {
+    fn push(&self, value: usize) {
+        unsafe { self.push(value) };
+    }
+
+    fn pop(&self) -> Option<usize> {
+        unsafe { self.pop() }
+    }
+}
+
+impl<const CAPACITY: usize> PushPop for FixedSizeSafelyOverflowingIndexQueue<CAPACITY> {
+    fn push(&self, value: usize) {
+        unsafe { self.push(value) };
+    }
+
+    fn pop(&self) -> Option<usize> {
+        unsafe { self.pop() }
+    }
+}
+
+fn perform_benchmark<Q: PushPop>(
+    args: &Args,
+    queue_a2b: Q,
+    queue_b2a: Q,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let barrier_handle = BarrierHandle::new();
+    let barrier = BarrierBuilder::new(3).create(&barrier_handle).unwrap();
+
+    let t1 = ThreadBuilder::new()
+        .affinity(args.cpu_core_participant_1)
+        .priority(255)
+        .spawn(|| {
+            barrier.wait();
+
+            for _ in 0..args.iterations {
+                queue_a2b.push(0);
+                while queue_b2a.pop().is_none() {}
+            }
+        });
+
+    let t2 = ThreadBuilder::new()
+        .affinity(args.cpu_core_participant_2)
+        .priority(255)
+        .spawn(|| {
+            barrier.wait();
+
+            for _ in 0..args.iterations {
+                while queue_a2b.pop().is_none() {}
+
+                queue_b2a.push(0);
+            }
+        });
+
+    std::thread::sleep(std::time::Duration::from_millis(100));
```

Comment:
This should be replaced with a barrier for setup. Just in case there are some weird systems where thread creation is really expensive